resources/image-document.html

```diff
@@ -0,0 +1,79 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title></title>
+    <style>
+      html,
+      body {
+        background-color: white;
+        margin: 0;
+        width: 100%;
+        height: 100%;
+      }
+
+      body {
+        display: flex;
+        justify-content: center;
+        align-items: center;
+      }
+
+      body.zoomed-in img {
+        cursor: zoom-out;
+      }
+
+      body.zoomed-out img {
+        cursor: zoom-in;
+        max-width: 100vw;
+        max-height: 100vh;
+      }
+
+      img {
+        /* Creates a checkerboard background for transparent images */
+        background: repeating-conic-gradient(#ccc 0% 25%, white 0% 50%);
+        background-size: 40px 40px;
+      }
+    </style>
+    <script>
+      // true if we try to adjust the image size to fit the viewport,
+      // false if we let it scroll when it's larger.
+      var adjustSize = true;
+
+      var img;
+
+      function configureImage() {
+        const body = document.body;
+        body.classList.remove("zoomed-in");
+        body.classList.remove("zoomed-out");
+
+        if (
+          img.naturalWidth > window.innerWidth ||
+          img.naturalHeight > window.innerHeight
+        ) {
+          body.classList.add(adjustSize ? "zoomed-out" : "zoomed-in");
+        }
```

Comment:
Avoid unnecessary DOM changes like removing a class that is present just to immediately add it back.

```suggestion
        let overflows = img.naturalWidth > window.innerWidth || img.naturalHeight > window.innerHeight;
        img.classList.toggle("zoomed-in", overflows && !adjustSize);
        img.classList.toggle("zoomed-out", overflows && adjustSize);
```

Reply:
This could also make sense:

```js
        window.onresize = img.onload = () => {
          img.classList.toggle(
            "too-big",
            img.naturalWidth > window.innerWidth || img.naturalHeight > window.innerHeight,
          );
        };
        img.onclick = () => {
          img.classList.toggle("adjust");
        };
```
```css
      img.too-big:not(.adjust) {
        cursor: zoom-out;
      }

      img.too-big.adjust {
        cursor: zoom-in;
        max-width: 100vw;
        max-height: 100vh;
      }
```

```html
<img src="#" class="adjust">
```